packages/front/src/routes/auth/activate/[token]/components/StructureStep/StructureStep.controller.ts

```diff
@@ -0,0 +1,63 @@
+import { AgentJobTypeEnum } from "@api-subventions-asso/dto";
+import Dispatch from "$lib/core/Dispatch";
+import Store from "$lib/core/Store";
+import { isPhoneNumber } from "$lib/helpers/stringHelper";
+
+type Option = {
+    value: AgentJobTypeEnum;
+    label: string;
+};
+
+export default class StructureStepController {
+    private readonly dispatch: (_: string) => void;
+    public readonly options: Option[];
+    public readonly errors: Store<{ [key: string]: string | undefined }>;
+    private static errorMandatory = "Ce champ est obligatoire";
+    private readonly validators: Record<string, (value: any) => string | undefined>;
+    private readonly dirty: { [key: string]: boolean };
+
+    constructor() {
+        this.dispatch = Dispatch.getDispatcher();
+        this.options = [
+            { value: AgentJobTypeEnum.ADMINISTRATOR, label: "Gestionnaire administratif et financier" },
+            {
+                value: AgentJobTypeEnum.EXPERT,
+                label: "Chargé de mission / Expert métier",
+            },
+            { value: AgentJobTypeEnum.SERVICE_HEAD, label: "Responsable de service" },
+            { value: AgentJobTypeEnum.OTHER, label: "Autre" },
+        ];
+        this.errors = new Store({});
+        this.validators = {
+            service: (text: string) => {
+                if (!text) return StructureStepController.errorMandatory;
+            },
+            jobType: (jobs: string[] | null) => {
+                if (!jobs?.length) return StructureStepController.errorMandatory;
+            },
+            phoneNumber: (number: string) => {
+                if (!number) return StructureStepController.errorMandatory;
+                if (!isPhoneNumber(number)) return "Entrez un numéro de téléphone valide";
+            },
+        };
+        this.dirty = {
+            service: false,
+            jobType: false,
+            phoneNumber: false,
+        };
+    }
+
+    onUpdate(values: Record<string, unknown>, changedKey: string) {
+        this.dirty[changedKey] = true;
+        const tempErrors: { [key: string]: string | undefined } = {};
+        let currentError;
+        let error = false;
+        for (const [key, validator] of Object.entries(this.validators)) {
+            currentError = validator(values[key] as string);
```

Comment:
pourquoi ne pas faire directement `error = validator()` ?

Reply:
Je comprends pas bien ce que tu voudrais que je fasse. Le but de ça c'est de valider tous les champs et de faire la différenciation entre afficher un message d'erreur ou seulement bloquer silencieusement l'étape si l'utilisateur n'a jamais validé le champ

Reply:
Ce que je veux dire c'est que `let error` ne sert à priori à rien puisque tu as également l'info dans `currentError` qui si j'ai bien compris est `undefined` en cas de champ valide. Donc au lieu de faire `if (currentError) error = true` dans la boucle tu peux simplement faire un `if (this.errors.value.find(error => error)` ou un truc du genre pour savoir si le formulaire est en erreur. Mais ça vaudrait le coup d'en discuter parce que je pense que soit 1) je comprend pas bien soit 2) c'est trop compliqué

Reply:
J'ai renommé la variable et mis un commentaire pour clarifier